src/node_buffer.cc

```diff
@@ -616,7 +616,7 @@ void Fill(const FunctionCallbackInfo<Value>& args) {
       enc == UCS2 ? str_obj->Length() * sizeof(uint16_t) : str_obj->Length();
 
   if (enc == HEX && str_length  % 2 != 0)
-    return env->ThrowTypeError("Invalid hex string");
+    return env->ThrowError("Invalid hex string length");
```

Comment:
Maybe we can be a bit more explicit about it? Something like `String length must be a multiple of 2 if encoding is "hex"`?